go-controller/pkg/ovn/egressip.go

```diff
@@ -1670,39 +1712,71 @@ func (e *egressIPZoneController) getTransitIP(nodeName string, wantsIPv6 bool) (
 	return nodeTransitIP.IP.String(), nil
 }
 
-// createReroutePolicyOps creates an operation that does idempotent updates of the
-// LogicalRouterPolicy corresponding to the egressIP status item, according to the
-// following update procedure:
-// - if the LogicalRouterPolicy does not exist: it adds it by creating the
-// reference to it from ovn_cluster_router and specifying the array of nexthops
-// to equal [gatewayRouterIP]
-// - if the LogicalRouterPolicy does exist: it adds the gatewayRouterIP to the
-// array of nexthops
-// This function should be called with lock on nodeZoneState cache key status.Node
-func (e *egressIPZoneController) createReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName string) ([]ovsdb.Operation, error) {
-	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+func (e *egressIPZoneController) getNextHop(egressNodeName, egressIP, egressIPName string, isLocalZoneEgressNode, isOVNManaged bool) (string, error) {
```

Comment:
some doc text on this function would be good

Reply:
Done